ltdiy.go

```diff
@@ -8,206 +8,206 @@
 type lineUpdate struct {
-    LineID string
-    Index int
-    Times []int
+	LineID string
+	Index  int
+	Times  []int
 }
 
 type stationUpdate struct {
-    StationID string
-    Lines []lineUpdate
+	StationID string
+	Lines     []lineUpdate
 }
 
 type update struct {
-    Stops []stationUpdate
+	Stops []stationUpdate
 }
 
-
 // This is the main system information; fill this as appropriate
-var mainSystem system = system {
-    Name: "TEECOM Shuttle Service",
-    Tagline: "San Francisco Bay Area",
-    Stops: []station {
-        station {
-            Name: "TEECOM",
-            ID: "tee",
-            Coord: coordinates {
-                Lat: 10.5,
-                Lon: 10.5,
-            },
-            Directions: [2]string {"Northbound", "Southbound"},
-            Lines: [2]map[string]*line {
-                map[string]*line {
-                    "sh": &line {
-                        Name: "Shuttle",
-                        ID: "sh",
-                        Color: "#ff0000",
-                    },
-                },
-            },
-        },
-    },
-    
-    TimeMax: 45,
-    
-    stopMap: make(map[string]*station),
+var mainSystem system = system{
+	Name:    "TEECOM Shuttle Service",
+	Tagline: "San Francisco Bay Area",
+	Stops: []station{
+		station{
+			Name: "TEECOM",
+			ID:   "tee",
+			Coord: coordinates{
+				Lat: 10.5,
+				Lon: 10.5,
+			},
+			Directions: [2]string{"Northbound", "Southbound"},
+			Lines: [2]map[string]*line{
+				map[string]*line{
+					"sh": &line{
+						Name:  "Shuttle",
+						ID:    "sh",
+						Color: "#ff0000",
+					},
+				},
+			},
+		},
+	},
+
+	TimeMax: 45,
+
+	stopMap: make(map[string]*station),
 }
 
 // Where static files will be found
 const staticDirectory string = "static"
 
 func prepareSystem() {
-    // Cache system IDs for future lookup
-    for i := 0; i < len(mainSystem.Stops); i++ {
-        stop := &mainSystem.Stops[i]
-        mainSystem.stopMap[stop.ID] = stop
-    }
+	// Cache system IDs for future lookup
+	for i := 0; i < len(mainSystem.Stops); i++ {
+		stop := &mainSystem.Stops[i]
+		mainSystem.stopMap[stop.ID] = stop
+	}
 }
 
 func main() {
-    log.Println("Starting server")
-    prepareSystem()
-    
-    // Setup routing
-    http.HandleFunc("/info", handleInfo)
-    http.HandleFunc("/update", handleUpdate)
-    http.HandleFunc("/stop", handleStopInfo)
-    
-    // Run server on port 8080
-    http.ListenAndServe(":8080", nil)
+	log.Println("Starting server")
+	prepareSystem()
+
+	// Setup routing
+	http.HandleFunc("/info", handleInfo)
+	http.HandleFunc("/update", handleUpdate)
+	http.HandleFunc("/stop", handleStopInfo)
+
+	// Run server on port 8080
+	if err := http.ListenAndServe(":8080", nil); err != nil {
+		log.Fatal(err)
+	}
 }
 
 // JSON encode all of the information
 func handleInfo(w http.ResponseWriter, r *http.Request) {
-    js := json.NewEncoder(w)
-    js.Encode(mainSystem)
+	if err := json.NewEncoder(w).Encode(mainSystem); err != nil {
```

Comment:
This also uses Go's conditional with setup, which is really handy for checking errors! 😄